pkg/crd/markers/crd.go

```diff
@@ -316,3 +319,29 @@ func (s UnservedVersion) ApplyToCRD(crd *apiext.CustomResourceDefinitionSpec, ve
 }
 
 // NB(directxman12): singular was historically distinct, so we keep it here for backwards compat
+
+// +controllertools:marker:generateHelp:category=CRD
+
+// DeprecatedVersion marks this version as deprecated.
+type DeprecatedVersion struct {
+	// Warning message to be shown on the deprecated version
+	Warning string
```

Comment:
is this optional, or is it required in the actual CRD.  If optional, you can make it a pointer or do something like

```go
type DeprecatedVersion struct {
   Warning string `marker:",optional"`
}
```

So that the equivalent field is also optional, meaning either of the following would work:

```go
// +kubebuilder:deprecated
type SomeKind struct {}

// {  deprecated: true }
```

```go
// +kubebuilder:deprecated:warning="blah"
type SomeKind struct{}

// { deprecated: true, deprecationWarning: "blah" }
```

Reply:
it's optional, but the CRD for CRD also takes a pointer for the deprecation message, so I added the marker optional and also made it a pointer